std/uni.d

```diff
@@ -8214,6 +8215,162 @@ unittest //12428
     assert(s == "abcdefghij");
 }
 
+
+// generic toUpper/toLower on whole range, returns range
+private auto toCaser(alias indexFn, uint maxIdx, alias tableFn, Range)(Range str)
+    if (isInputRange!Range && isSomeChar!(ElementEncodingType!Range))
+{
+    static if (!is(Unqual!(ElementEncodingType!Range) : const(dchar)))
```

Comment:
I personally don't think the discrepancy is worth it. surely autodecoding doesn't *have to* allocate and doesn't *have to* be slow. The general consensus is that we need to solve `@nogc` for exceptions in general, so the only necessary difference I can see between the two approaches is that one throws while the other uses a replacement character. As shown in the example for [std.exception.handle](http://dlang.org/phobos/std_exception#handle), it's possible to get replacement characters with autodecoding as well, and it can be made nothrow with an additional call to `handle`.

Chances are if you have a range composition using the "new style" decoding, you'll be mixing it with something using "old style" decoding, so to get `nothrow` you need to use `handle` anyway.